server/app/routes/review/index.js

```diff
@@ -0,0 +1,58 @@
+'use strict'
+
+var router = require('express').Router();
+var mongoose = require('mongoose');
+module.exports = router;
+require('../../../db/models');
+var Review = mongoose.model('Review');
+
+
+router.get('/', function (req, res, next) {
+  Review.find({}).exec()
+    .then(function (reviews) {
+      res.status(200).json(reviews);
+    })
+    .then(null, function (err) {
+      console.error("You got an error", err);
+    })
+    .then(null, next);
+});
+
+router.get('/:id', function (req, res, next) {
+  //change user to product for product routes
+  Review.find({user: req.params.id})
+  .then(function (reviews) {
+    res.status(200).json(reviews);
+  })
+  .then(null, next);
+});
+
+router.post('/', function (req, res, next) {
+  Review.create(req.body)
+    .then(function (review) {
+      res.status(201).json(review);
+    })
+    .then(null, next);
+});
+
+router.put('/:id', function (req, res, next) {
+  Review.findOne({_id: req.params.id})
+    .then(function (review) {
+      for (var k in req.body) {
+        review[k] = req.body[k];
+      }
+      review.save();
+      res.status(200).json(review);
```

Comment:
We're not waiting for the old review to save to the database fully before sending it back to the user. That means the copy the user receives is different from the updated version.